tools/desktop/desktop.go

```diff
@@ -0,0 +1,230 @@
+	if err == nil {
+		return
+	}
+	fmt.Fprintf(c.App.ErrWriter, "Error: %+v\n", err)
+	cli.OsExiter(1)
+}
+
+func createApp(reader io.Reader, writer io.Writer, exitErrHandler cli.ExitErrHandlerFunc) *cli.App {
+	app := cli.NewApp()
+	app.Name = "desktop"
+	app.Usage = "Tool to generate the Fleet Desktop application"
+	app.ExitErrHandler = exitErrHandler
+	cli.VersionPrinter = func(c *cli.Context) {
+		version.PrintFull()
+	}
+	app.Reader = reader
+	app.Writer = writer
+	app.ErrWriter = writer
+	app.Flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:    "version",
+			Usage:   "Version of the Fleet Desktop application",
+			EnvVars: []string{"FLEET_DESKTOP_VERSION"},
+		},
+		&cli.BoolFlag{
+			Name:    "verbose",
+			Usage:   "Log detailed information when building the application",
+			EnvVars: []string{"FLEET_DESKTOP_VERBOSE"},
+		},
+	}
+
+	app.Commands = []*cli.Command{
+		macos(),
+	}
+	return app
+}
+
+func macos() *cli.Command {
+	return &cli.Command{
+		Name:        "macos",
+		Usage:       "Creates the Fleet Desktop Application for macOS",
+		Description: "Builds and signs the Fleet Desktop .app bundle for macOS",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "authority",
+				Usage:   "Authority to use on the codesign invocation (if not set, app is not signed)",
+				EnvVars: []string{"FLEET_DESKTOP_APPLE_AUTHORITY"},
+			},
+		},
+		Action: func(c *cli.Context) error {
+			if !c.Bool("verbose") {
+				zlog.Logger = zerolog.Nop()
+			}
+			return createMacOSApp(c.String("version"), c.String("authority"))
+		},
+	}
+}
+
+func createMacOSApp(version, authority string) error {
+	const (
+		appDir           = "Fleet Desktop.app"
+		bundleIdentifier = "com.fleetdm.desktop"
+		// infoPList is the Info.plist file to use for the macOS .app bundle.
+		//
+		// 	- NSHighResolutionCapable=true: avoid having a blurry icon and text.
+		//	- LSUIElement=1: avoid showing the app on the Dock.
+		infoPList = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleExecutable</key>
+	<string>fleet-desktop</string>
+	<key>CFBundleIdentifier</key>
+	<string>%s</string>
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>6.0</string>
+	<key>CFBundleName</key>
+	<string>fleet-desktop</string>
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>%s</string>
+	<key>CFBundleVersion</key>
+	<string>%s</string>
+	<key>NSHighResolutionCapable</key>
+	<string>True</string>
+	<key>LSUIElement</key>
+	<string>1</string>
+</dict>
+</plist>
+`
+	)
+
+	if runtime.GOOS != "darwin" {
+		return errors.New("The \"Fleet Desktop\" macOS app can only be created from macOS")
+	}
+
+	defer os.RemoveAll(appDir)
+
+	contentsDir := filepath.Join(appDir, "Contents")
+	macOSDir := filepath.Join(contentsDir, "MacOS")
+	for _, dir := range []string{macOSDir} {
```

Comment:
Is that because you expect to add more soon? Or should `contentsDir` be part of the slice?